openpathsampling/netcdfplus/dictify.py

```diff
@@ -205,7 +211,7 @@ def to_json(self, obj, base_type=''):
         return json.dumps(simplified)
 
     def to_json_object(self, obj):
-        if hasattr(obj, 'base_cls'):
+        if hasattr(obj, 'base_cls') and type(obj) is not type and type(obj) is not abc.ABCMeta:
```

Comment:
I see. The underlying idea was that a store (the class instances that stores objects) should be storable objects themselves. In doing so the stored file contains most of the structure in it, not only the data. I think that was clear. Now to save a store I needed to add a new variable type for the to_json functions. The thing that I named simplifier. It takes a dict (presumable the one from `to_dict`) and converts all objects that cannot directly turned into json into a little dict that will be reconstructed when loading. So 
```
object >(to_dict())> dict >(simplifier))> simple_dict >(json.decode())> string
```
The simplifier never needed to encode a class itself. And my implementation checked for `type(obj) == type` and if the class was subclassed of storableobjects (we do not allow storing unknown classes). In that case we take the name of the class and store it with a `_class` key. When reconstructing it realizes the `_class` key and picks the appropriate class from the list of registered storable objects. 

Now to your question. The point is that before it just never occurred that an instance had an attribute that pointed to a class. Now for storing stores themselves the store has an attribute that points to a class. Namely the base class to be stored, which I call content_class.

Hence the need to update the simplifier to handle now abstract classes as well. This would have failed before too, but since we never stored abstract classes the error never showed up.